src/modules/adaptors/Convex/ConvexCurveAdaptor.sol

```diff
@@ -0,0 +1,333 @@
+     * @param _baseRewardPool for respective convex market (w/ trusted poolId)
+     * @param _amount of LPTs to unstake, unwrap, and withdraw from convex market to calling cellar
+     * @param _claim whether or not to claim all rewards from BaseRewardPool
+     */
+    function withdrawFromBaseRewardPoolAsLPT(address _baseRewardPool, uint256 _amount, bool _claim) public {
+        IBaseRewardPool baseRewardPool = IBaseRewardPool(_baseRewardPool);
+
+        if (_amount == type(uint256).max) {
+            _amount = baseRewardPool.balanceOf(address(this));
+        }
+        baseRewardPool.withdrawAndUnwrap(_amount, _claim);
+    }
+
+    /**
+     * @notice Allows strategists to get rewards for an Convex Booster without withdrawing/unwrapping from Convex market
+     * @param _pid specified pool ID corresponding to LPT convex market
+     * @param _baseRewardPool for respective convex market (w/ trusted poolId)
+     * @param _claimExtras Whether or not to claim extra rewards associated to the Convex booster (outside of rewardToken for Convex booster)
+     */
+    function getRewards(
+        uint256 _pid,
+        address _baseRewardPool,
+        bool _claimExtras,
+        ERC20 _lpt,
+        CurvePool _pool,
+        bytes4 _selector
+    ) public {
+        _validatePositionIsUsed(_pid, _baseRewardPool, _lpt, _pool, _selector);
+        _getRewards(_baseRewardPool, _claimExtras);
+    }
+
+    /**
+     * @notice Validates that a given pid (poolId), and baseRewardPool are set up as a position with this adaptor in the calling Cellar.
+     * @dev This function uses `address(this)` as the address of the calling Cellar.
+     */
+    function _validatePositionIsUsed(
+        uint256 _pid,
+        address _baseRewardPool,
+        ERC20 _lpt,
+        CurvePool _curvePool,
+        bytes4 _selector
+    ) internal view {
+        bytes32 positionHash = keccak256(
+            abi.encode(identifier(), false, abi.encode(_pid, _baseRewardPool, _lpt, _curvePool, _selector))
+        );
+        uint32 positionId = Cellar(address(this)).registry().getPositionHashToPositionId(positionHash);
+        if (!Cellar(address(this)).isPositionUsed(positionId))
+            revert ConvexAdaptor__ConvexBoosterPositionsMustBeTracked(
+                _pid,
+                _baseRewardPool,
+                _lpt,
+                _curvePool,
+                _selector
+            );
+    }
+
+    /**
+     * @notice Call a reentrancy protected function in `pool`.
+     * @dev Used to insure `pool` is not in a manipulated state.
+     */
+    function _callReentrancyFunction(CurvePool pool, bytes4 selector) internal {
```

Comment:
After merge from dev, use the CurveHelper for this function, and the associated error

Reply:
Roger that, making the change.